src/tvdb/client.py

```diff
@@ -212,12 +215,31 @@ class Series(_Media):
     ResponseType = SeriesIdGetResponse
     ExtendedResponseType = SeriesIdExtendedGetResponse
 
+    def __init__(self, client: "TvdbClient", data: AnyRecord | SearchResult | None):
+        super().__init__(client, data)
+        self.episodes: list[EpisodeBaseRecord] | None = None
+        self.seasons: list[SeasonBaseRecord] | None = None
+        if isinstance(self.data, SeriesExtendedRecord):
+            self.seasons = self.data.seasons
+
     @override
     @classmethod
     async def supports_meta(cls, meta: FetchMeta) -> bool:
         """Check if the class supports a specific meta."""
         return meta in {FetchMeta.TRANSLATIONS, FetchMeta.EPISODES}
 
+    async def find_episodes(self, session: aiohttp.ClientSession, season_type: str = "official") -> None:
+        """Fetch episodes for the series based on the season type."""
+        self.client.http_session = session
+        endpoint = f"series/{self.id}/episodes/{season_type}"
+        response = await self.client.request("GET", endpoint)
+
+        # Assuming 'episodes' field contains the list of episodes
+        response = SeriesIdEpisodesSeasonTypeGetResponse(**response)  # pyright: ignore[reportCallIssue]
+
+        if response.data:
+            self.episodes = response.data.episodes
```

Comment:
The only thing here is that it `Series` class needs `data` when it's initialized, which means it's impossible to fetch an episode without having previously fetched a series, even if you know the series id and don't need the data on the entire series.

One way to solve this would be to give episodes their own class, and make this a classmethod that returns a list of these Episode classes. From Series, this method could stay, but it will just call `Episode.find_episodes(self.id, season_type)`.